conda/common/io.py

```diff
@@ -689,6 +693,130 @@ def _ensure_dir(self):
             os.makedirs(dirname(self.record_file))
 
 
+Reporter = Dict[str, Union[bool, int, str]]
+DetailRecord = Dict[str, Union[str, int, bool]]
+
+
+class ReporterHandlerBase(ABC):
+    """
+    Base class for all reporter handlers
+    """
+
+    @abstractmethod
+    def detail_view(self, data: DetailRecord, **kwargs) -> str:
+        """
+        This method is responsible for generating the output in a "tabular" format.
+        """
+        ...
+
+    @abstractmethod
+    def string_view(self, data: str, **kwargs) -> str:
+        """
+        This method returns simple string output.
+        """
+        ...
+
+
+class ConsoleHandler(ReporterHandlerBase):
+    """
+    Reporter handler for standard output (classic output).
+    """
+
+    def detail_view(self, data: DetailRecord, **kwargs) -> str:
+        table_str = ""
+        longest_header = len(sorted(data.keys(), key=len).pop())
+
+        for header, value in data.items():
+            row_header = header.ljust(longest_header, " ")
+            table_str += f"{row_header} : {value}\n"
```

Comment:
```suggestion
            table_str += f"{row_header:<{longest_header}} : {value}\n"
```

Reply:
If the suggestion above looks magical take a look at https://docs.python.org/3/library/string.html#format-specification-mini-language for more details on the format string mini-language:

<img width="819" alt="Screenshot 2024-04-24 at 17 52 27" src="https://github.com/conda/conda/assets/4546435/c1777e15-eda7-4a18-8312-206043c1667e">